Apps/GatewayApi/src/Controllers/PhsaController.cs

```diff
@@ -0,0 +1,89 @@
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+
+    /// <summary>
+    /// Web API to handle dependent interactions.
+    /// </summary>
+    [Authorize]
+    [ApiVersion("1.0")]
+    [Route("[controller]")]
+    [ApiController]
+    public class PhsaController
+    {
+        private readonly IDependentService dependentService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PhsaController"/> class.
+        /// </summary>
+        /// <param name="dependentService">The injected dependent service.</param>
+        public PhsaController(IDependentService dependentService)
+        {
+            this.dependentService = dependentService;
+        }
+
+        /// <summary>
+        /// Gets all dependents for the specified user.
+        /// </summary>
+        /// <returns>The list of dependent model wrapped in a request result.</returns>
+        /// <param name="hdid">The owner hdid.</param>
+        /// <response code="200">Returns the list of dependents.</response>
+        /// <response code="401">the client must authenticate itself to get the requested response.</response>
+        /// <response code="403">
+        /// The client does not have access rights to the content; that is, it is unauthorized, so the server
+        /// is refusing to give the requested resource. Unlike 401, the client's identity is known to the server.
+        /// </response>
+        [HttpGet]
+        [Authorize(Policy = UserProfilePolicy.Read)]
+        [Route("dependents/{hdid}")]
+        public RequestResult<IEnumerable<DependentModel>> GetAll(string hdid)
+        {
+            return this.dependentService.GetDependents(hdid);
+        }
+
+        /// <summary>
+        /// Gets all dependents for the specified date range.
+        /// </summary>
+        /// <returns>The list of dependents wrapped in a request result.</returns>
+        /// <param name="fromDateUtc">The from date in Utc. Required.</param>
+        /// <param name="toDateUtc">The to date in Utc.</param>
+        /// <param name="pageNumber">The page number.</param>
+        /// <param name="pageSize">The page size. Max 5000.</param>
+        /// <response code="200">Returns the list of dependents.</response>
+        /// <response code="401">the client must authenticate itself to get the requested response.</response>
+        /// <response code="403">
+        /// The client does not have access rights to the content; that is, it is unauthorized, so the server
+        /// is refusing to give the requested resource. Unlike 401, the client's identity is known to the server.
+        /// </response>
+        [HttpGet]
+        [Authorize(Policy = UserProfilePolicy.Read)]
+        [Route("dependents")]
+        public RequestResult<IEnumerable<GetDependentResponse>> GetAll([FromQuery] DateTime fromDateUtc, [FromQuery] DateTime toDateUtc, [FromQuery] int pageNumber, [FromQuery] int pageSize)
```

Comment:
Do we want to make todate, page number nad page size mandatory? if they are optional, they should probably be nullable